csharp/Antlr4cs/CSharpLexerBase.cs

```diff
@@ -0,0 +1,106 @@
+using Antlr4.Runtime;
+using System.Collections.Generic;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+public abstract class CSharpLexerBase : Lexer
+{
+    public CSharpLexerBase(ICharStream input)
+        : base(input)
+    {
+    }
+
+    protected int interpolatedStringLevel;
+    protected Stack<bool> interpolatedVerbatiums = new Stack<bool>();
+    protected Stack<int> curlyLevels = new Stack<int>();
+    protected bool verbatium;
+
+    protected void Action1()
```

Comment:
I think it should be more meaningful names for actions. `ActionN` does not say what action really does. For instance, `IncrementInterpolatedStringLevel`.

Reply:
Yeah, I agree. I was just lazy. I'll do something like "On<Rule name>" instead.